platforms/android/library/src/main/java/io/element/android/wysiwyg/EditorEditText.kt

```diff
@@ -66,39 +64,27 @@ class EditorEditText : TextInputEditText {
         selectionChangeListener?.selectionChanged(selStart, selEnd)
     }
 
-    /**
-     * We wrap the internal [EditableInputConnection] as it's not public, but its internal behavior
-     * is probably needed to work properly with the EditText.
-     */
     override fun onCreateInputConnection(outAttrs: EditorInfo): InputConnection {
-        val baseInputConnection = requireNotNull(super.onCreateInputConnection(outAttrs))
         val inputConnection =
-            InterceptInputConnection(baseInputConnection, this, inputProcessor)
+            InterceptInputConnection(this, inputProcessor)
         this.inputConnection = inputConnection
         return inputConnection
     }
 
-    private fun addHardwareKeyInterceptor() {
-        // This seems to be the only way to prevent EditText from automatically handling key strokes
-        setOnKeyListener { v, keyCode, event ->
-            if (event.keyCode == KeyEvent.KEYCODE_ENTER) {
-                if (event.action == MotionEvent.ACTION_DOWN) {
-                    inputConnection?.sendHardwareKeyboardInput(event)
-                }
-                true
-            } else if (event.action != MotionEvent.ACTION_DOWN) {
-                false
-            } else if (event.isMovementKey()) {
-                false
-            } else if (event.metaState != 0 && event.unicodeChar == 0) {
-                // Is a modifier key
-                false
-            } else {
-                inputConnection?.sendHardwareKeyboardInput(event)
-                true
-            }
+
+    override fun onKeyDown(keyCode: Int, event: KeyEvent): Boolean =
```

Comment:
According to [the docs](https://developer.android.com/reference/android/widget/TextView#onKeyDown(int,%20android.view.KeyEvent)) this is not a reliable source of software key events.